python/lsst/ci/prepare.py

```diff
@@ -356,15 +322,15 @@ def _fetch(self, product):
                     # lfs will pickup the .gitconfig and pull lfs objects for
                     # the default ref during clone.  Config options set on the
                     # cli during the clone get recorded in `.git/config'
-                    args += ['-c', 'filter.lfs.required']
-                    args += ['-c', 'filter.lfs.smudge=git-lfs smudge %f']
+                    args += ['-c', 'filter.lfs.required=false']
+                    args += ['-c', 'filter.lfs.smudge=']
```

Comment:
Yes - this allows us to defer the LFS file download.

This is important for expected future modifications of the code because all LFS repos (in lsst_distrib specifically) are `setupOptional`, and may not be required for a particular build, but we still need to perform the git clone to see what the git commit was when calculating the hash-based version.

However, on a second look, I think I should consider reconfiguring these configurations after the initial git-lfs-pull, when it has occurred.

Reply:
I added code to reconfigure the repos as they were configured before _after_ the git-lfs pull has been executed, in commit 284893c. This technically was not necessary if a user is using lsst_build to manage their repos, but this results in the same local git-repo configuration as before, which is almost always the case.